src/components/picker/testing/index.tsx

```diff
@@ -1,35 +1,109 @@
-import { Component, h, State } from "@stencil/core"
+import { Component, h, Host, Listen, State } from "@stencil/core"
+import { Data, Notice } from "../../../model"
+
+function validateEmail(email: string) {
+	return email.match(/^.+@.+/) ? true : { result: false, notice: Notice.failed("That is not an email") }
+}
 
 @Component({
-	tag: "smoothly-picker-tester",
+	tag: "smoothly-picker-demo",
 	styleUrl: "style.css",
 	scoped: true,
 })
-export class SmoothlyPickerTester {
+export class SmoothlyPickerDemo {
+	@State() readonly = false
+	private users: Record<string, string | undefined> = {
+		"giovani@rocket.com": "giovani doe",
+		"jessie@rocket.com": "jessie doe",
+		"james@rocket.com": "james doe",
+	}
 	@State() data = {
 		message: "hello world",
 		emails: ["giovani@rocket.com", "jessie@rocket.com", "james@rocket.com"],
 	}
+	inputHandler(event: CustomEvent<Data>) {
+		this.data = { ...this.data, ...event.detail }
+		console.log(this.data)
+	}
+	@Listen("smoothlyFormSubmit")
+	submitHandler(event: CustomEvent<Data>) {
+		console.log("submitted", event.detail)
+	}
 	render() {
 		return (
-			<smoothly-form
-				looks="grid"
-				onSmoothlyFormInput={e => (this.data = { ...this.data, ...e.detail })}
-				onSmoothlyFormSubmit={e => console.log("submit", { ...this.data, ...e.detail })}>
-				<smoothly-input type="text" name="message" value={this.data.message}>
-					Message
-				</smoothly-input>
-				<smoothly-picker name="type" multiple mutable>
-					<span slot="label">Emails</span>
-					<span slot="search">Search</span>
-					{this.data.emails.map(email => (
-						<smoothly-picker-option value={email} selected>
-							{email}
+			<Host>
+				<smoothly-button onClick={() => (this.readonly = !this.readonly)}>
+					{this.readonly ? "current readonly" : "current writeable"}
```

Comment:
The effect of this butto ndoes not work all the time. the Picker gets stuck on readonly when toggling.

Reply:
the one thing the PR was set out to solve :joy: 
that must have been the worst places `if` of my life.